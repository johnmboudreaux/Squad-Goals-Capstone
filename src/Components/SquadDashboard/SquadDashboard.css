.squad-dashboard-button {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%; }

.squad-dashboard-button input {
  background: none;
  border: none;
  color: #d2d3d3;
  font-size: 1.3em;
  font-weight: lighter;
  text-decoration: none; }

.squad-dashboard-button a {
  color: #d2d3d3;
  font-size: 1.4em;
  font-weight: lighter;
  text-decoration: none; }
