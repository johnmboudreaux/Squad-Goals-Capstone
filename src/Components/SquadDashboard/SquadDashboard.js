import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Conversation from '../Conversation/Conversation';
import SubmitComment from '../SubmitComment/SubmitComment';

class SquadDashboard extends Component {
  constructor() {
    super();
    this.state = {
      squad: {},
      activeUser: false,
    };
  }

  componentDidMount() {
    this.getSquadData();
  }

  getSquadData() {
    const url = `/api/v1/squads/${this.props.match.params.id}`;
    return this.contentFetch(url)
      .then((parsedData) => {
        this.setState({ squad: parsedData });
        this.checkForEnrolledUser();
      })
      .catch(error => console.error(error));
  }

  checkForEnrolledUser() {
    const { userId } = this.props.location.state;
    const squadId = parseInt(this.props.match.params.id);
    const url = `/api/v1/users/${userId}/squads/${squadId}`;
    return fetch(url)
      .then(result => result.json())
      .then((response) => {
        if (response.length === 1) {
          this.setState({ activeUser: true });
        }
      })
      .catch(error => console.error(error));
  }

  joinSquad() {
    const { userId } = this.props.location.state;
    const squadId = parseInt(this.props.match.params.id);
    const postBody = {
      user_id: userId,
      squad_id: squadId,
    };
    fetch(`/api/v1/users/${userId}/squads/${squadId}`, {
      method: 'POST',
      body: JSON.stringify(postBody),
      headers: {
        'Content-Type': 'application/json',
      },
    })
      .then(response => response)
      .catch(error => console.error(error));
  }

  leaveSquad() {
    const { userId } = this.props.location.state;
    const squadId = parseInt(this.props.match.params.id);
    fetch(`/api/v1/users/${userId}/squads/${squadId}`, {
      method: 'DELETE',
    })
      .then(response => response)
      .catch(error => console.error(error));
  }

  contentFetch(url) {
    return fetch(url)
      .then(result => result.json())
      .catch(error => console.error(error));
  }

  render() {
    return (
      <div className="dashboard-container">
        <div className="dashboard-body">
          <h1 className="dashboard-body-title">{this.state.squad.squad_name}</h1>
          <div className="dashboard-body-info">
            <div className="squad-dashbord-body-titles">
              <div>
                <h2>{this.state.squad.squad_name}</h2>
              </div>
              <div>
                <h2>Members</h2>
              </div>
              <div>
                <h2>Leaderboard</h2>
              </div>
              <div>
                <h2>Upcoming Goals</h2>
              </div>
            </div>
            <div className="squad-dashboard-button">
              <div>
                <Link to={{
                  pathname: '/CreateGoals',
                  state: { userId: this.props.location.state.userId },
                }}
                >
                New Goal
                </Link>
              </div>
              { this.state.activeUser ?
                <div>
                  <input
                    className=""
                    onClick={() => this.leaveSquad()}
                    type="submit"
                    value="Leave Squad"
                  />
                </div>
                :
                <div>
                  <input
                    className=""
                    onClick={() => this.joinSquad()}
                    type="submit"
                    value="Join A Squad"
                  />
                </div>
            }
            </div>
          </div>
          { this.state.squad.conversation !== undefined &&
            <div className="conversation-container">
              <Conversation comments={this.state.squad.conversation} />
              <SubmitComment conversationId={this.state.squad.conversation_id} />
            </div>
          }
        </div>
      </div>
    );
  }
}

SquadDashboard.propTypes = {
  match: PropTypes.object,
  params: PropTypes.object,
  location: PropTypes.object,
  state: PropTypes.object,
  id: PropTypes.string,
};

export default SquadDashboard;
