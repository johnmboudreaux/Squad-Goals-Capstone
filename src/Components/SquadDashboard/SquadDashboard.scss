@import '../../variables.scss';

.squad-dashboard-button {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%
}

.squad-dashboard-button input {
  background: none;
  border: none;
  color: $white;
  font-size: 1.3em;
  font-weight: lighter;
  text-decoration: none;
}

.squad-dashboard-button a {
  color: $white;
  font-size: 1.4em;
  font-weight: lighter;
  text-decoration: none;
}
