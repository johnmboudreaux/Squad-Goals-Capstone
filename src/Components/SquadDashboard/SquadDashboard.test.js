import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import SquadDashboard from './SquadDashboard';

describe('SquadDashboard snapshot', () => {
  const squad = {
    conversation: [],
    conversation_id: 24,
    id: 18,
    squad_name: 'Jm Squad Squad',
    users: [{}],
  };

  it('should always match the snapshot', () => {
    const wrapper = shallow(<SquadDashboard
      squad={squad}
      location={{ state: { userId: 1 } }}
      match={{ params: { id: 1 } }}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
