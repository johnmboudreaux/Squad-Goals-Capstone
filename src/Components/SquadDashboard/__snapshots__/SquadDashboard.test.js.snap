// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquadDashboard snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SquadDashboard
    location={
      Object {
        "state": Object {
          "userId": 1,
        },
      }
    }
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    squad={
      Object {
        "conversation": Array [],
        "conversation_id": 24,
        "id": 18,
        "squad_name": "Jm Squad Squad",
        "users": Array [
          Object {},
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="dashboard-body"
      >
        <h1
          className="dashboard-body-title goal-body-title"
        />
        <div
          className="dashboard-body-info"
        >
          <div
            className="squad-dashboard-body-titles"
          >
            <div>
              <h2 />
            </div>
            <div>
              <h2>
                Members
              </h2>
            </div>
            <div>
              <h2>
                Leaderboard
              </h2>
            </div>
            <div>
              <h2>
                Upcoming Goals
              </h2>
            </div>
          </div>
          <div
            className="squad-dashboard-button"
          >
            <div
              className="single-button-container"
            >
              <Link
                className="nav-button"
                replace={false}
                to={
                  Object {
                    "pathname": "/CreateGoals",
                    "state": Object {
                      "userId": 1,
                    },
                  }
                }
              >
                New Goal
              </Link>
            </div>
            <div
              className="single-button-container"
            >
              <input
                className="nav-button"
                onClick={[Function]}
                type="submit"
                value="Join Squad"
              />
            </div>
          </div>
        </div>
      </div>,
      "className": "dashboard-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="dashboard-body-title goal-body-title"
          />,
          <div
            className="dashboard-body-info"
          >
            <div
              className="squad-dashboard-body-titles"
            >
              <div>
                <h2 />
              </div>
              <div>
                <h2>
                  Members
                </h2>
              </div>
              <div>
                <h2>
                  Leaderboard
                </h2>
              </div>
              <div>
                <h2>
                  Upcoming Goals
                </h2>
              </div>
            </div>
            <div
              className="squad-dashboard-button"
            >
              <div
                className="single-button-container"
              >
                <Link
                  className="nav-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/CreateGoals",
                      "state": Object {
                        "userId": 1,
                      },
                    }
                  }
                >
                  New Goal
                </Link>
              </div>
              <div
                className="single-button-container"
              >
                <input
                  className="nav-button"
                  onClick={[Function]}
                  type="submit"
                  value="Join Squad"
                />
              </div>
            </div>
          </div>,
          false,
        ],
        "className": "dashboard-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "dashboard-body-title goal-body-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="squad-dashboard-body-titles"
              >
                <div>
                  <h2 />
                </div>
                <div>
                  <h2>
                    Members
                  </h2>
                </div>
                <div>
                  <h2>
                    Leaderboard
                  </h2>
                </div>
                <div>
                  <h2>
                    Upcoming Goals
                  </h2>
                </div>
              </div>,
              <div
                className="squad-dashboard-button"
              >
                <div
                  className="single-button-container"
                >
                  <Link
                    className="nav-button"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/CreateGoals",
                        "state": Object {
                          "userId": 1,
                        },
                      }
                    }
                  >
                    New Goal
                  </Link>
                </div>
                <div
                  className="single-button-container"
                >
                  <input
                    className="nav-button"
                    onClick={[Function]}
                    type="submit"
                    value="Join Squad"
                  />
                </div>
              </div>,
            ],
            "className": "dashboard-body-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <h2 />
                  </div>,
                  <div>
                    <h2>
                      Members
                    </h2>
                  </div>,
                  <div>
                    <h2>
                      Leaderboard
                    </h2>
                  </div>,
                  <div>
                    <h2>
                      Upcoming Goals
                    </h2>
                  </div>,
                ],
                "className": "squad-dashboard-body-titles",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h2 />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h2",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h2>
                      Members
                    </h2>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Members",
                    },
                    "ref": null,
                    "rendered": "Members",
                    "type": "h2",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h2>
                      Leaderboard
                    </h2>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Leaderboard",
                    },
                    "ref": null,
                    "rendered": "Leaderboard",
                    "type": "h2",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h2>
                      Upcoming Goals
                    </h2>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Upcoming Goals",
                    },
                    "ref": null,
                    "rendered": "Upcoming Goals",
                    "type": "h2",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="single-button-container"
                  >
                    <Link
                      className="nav-button"
                      replace={false}
                      to={
                        Object {
                          "pathname": "/CreateGoals",
                          "state": Object {
                            "userId": 1,
                          },
                        }
                      }
                    >
                      New Goal
                    </Link>
                  </div>,
                  <div
                    className="single-button-container"
                  >
                    <input
                      className="nav-button"
                      onClick={[Function]}
                      type="submit"
                      value="Join Squad"
                    />
                  </div>,
                ],
                "className": "squad-dashboard-button",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-button"
                      replace={false}
                      to={
                        Object {
                          "pathname": "/CreateGoals",
                          "state": Object {
                            "userId": 1,
                          },
                        }
                      }
                    >
                      New Goal
                    </Link>,
                    "className": "single-button-container",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "New Goal",
                      "className": "nav-button",
                      "replace": false,
                      "to": Object {
                        "pathname": "/CreateGoals",
                        "state": Object {
                          "userId": 1,
                        },
                      },
                    },
                    "ref": null,
                    "rendered": "New Goal",
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <input
                      className="nav-button"
                      onClick={[Function]}
                      type="submit"
                      value="Join Squad"
                    />,
                    "className": "single-button-container",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "nav-button",
                      "onClick": [Function],
                      "type": "submit",
                      "value": "Join Squad",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="dashboard-body"
        >
          <h1
            className="dashboard-body-title goal-body-title"
          />
          <div
            className="dashboard-body-info"
          >
            <div
              className="squad-dashboard-body-titles"
            >
              <div>
                <h2 />
              </div>
              <div>
                <h2>
                  Members
                </h2>
              </div>
              <div>
                <h2>
                  Leaderboard
                </h2>
              </div>
              <div>
                <h2>
                  Upcoming Goals
                </h2>
              </div>
            </div>
            <div
              className="squad-dashboard-button"
            >
              <div
                className="single-button-container"
              >
                <Link
                  className="nav-button"
                  replace={false}
                  to={
                    Object {
                      "pathname": "/CreateGoals",
                      "state": Object {
                        "userId": 1,
                      },
                    }
                  }
                >
                  New Goal
                </Link>
              </div>
              <div
                className="single-button-container"
              >
                <input
                  className="nav-button"
                  onClick={[Function]}
                  type="submit"
                  value="Join Squad"
                />
              </div>
            </div>
          </div>
        </div>,
        "className": "dashboard-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="dashboard-body-title goal-body-title"
            />,
            <div
              className="dashboard-body-info"
            >
              <div
                className="squad-dashboard-body-titles"
              >
                <div>
                  <h2 />
                </div>
                <div>
                  <h2>
                    Members
                  </h2>
                </div>
                <div>
                  <h2>
                    Leaderboard
                  </h2>
                </div>
                <div>
                  <h2>
                    Upcoming Goals
                  </h2>
                </div>
              </div>
              <div
                className="squad-dashboard-button"
              >
                <div
                  className="single-button-container"
                >
                  <Link
                    className="nav-button"
                    replace={false}
                    to={
                      Object {
                        "pathname": "/CreateGoals",
                        "state": Object {
                          "userId": 1,
                        },
                      }
                    }
                  >
                    New Goal
                  </Link>
                </div>
                <div
                  className="single-button-container"
                >
                  <input
                    className="nav-button"
                    onClick={[Function]}
                    type="submit"
                    value="Join Squad"
                  />
                </div>
              </div>
            </div>,
            false,
          ],
          "className": "dashboard-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "dashboard-body-title goal-body-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="squad-dashboard-body-titles"
                >
                  <div>
                    <h2 />
                  </div>
                  <div>
                    <h2>
                      Members
                    </h2>
                  </div>
                  <div>
                    <h2>
                      Leaderboard
                    </h2>
                  </div>
                  <div>
                    <h2>
                      Upcoming Goals
                    </h2>
                  </div>
                </div>,
                <div
                  className="squad-dashboard-button"
                >
                  <div
                    className="single-button-container"
                  >
                    <Link
                      className="nav-button"
                      replace={false}
                      to={
                        Object {
                          "pathname": "/CreateGoals",
                          "state": Object {
                            "userId": 1,
                          },
                        }
                      }
                    >
                      New Goal
                    </Link>
                  </div>
                  <div
                    className="single-button-container"
                  >
                    <input
                      className="nav-button"
                      onClick={[Function]}
                      type="submit"
                      value="Join Squad"
                    />
                  </div>
                </div>,
              ],
              "className": "dashboard-body-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      <h2 />
                    </div>,
                    <div>
                      <h2>
                        Members
                      </h2>
                    </div>,
                    <div>
                      <h2>
                        Leaderboard
                      </h2>
                    </div>,
                    <div>
                      <h2>
                        Upcoming Goals
                      </h2>
                    </div>,
                  ],
                  "className": "squad-dashboard-body-titles",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h2 />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "h2",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h2>
                        Members
                      </h2>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Members",
                      },
                      "ref": null,
                      "rendered": "Members",
                      "type": "h2",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h2>
                        Leaderboard
                      </h2>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Leaderboard",
                      },
                      "ref": null,
                      "rendered": "Leaderboard",
                      "type": "h2",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h2>
                        Upcoming Goals
                      </h2>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Upcoming Goals",
                      },
                      "ref": null,
                      "rendered": "Upcoming Goals",
                      "type": "h2",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="single-button-container"
                    >
                      <Link
                        className="nav-button"
                        replace={false}
                        to={
                          Object {
                            "pathname": "/CreateGoals",
                            "state": Object {
                              "userId": 1,
                            },
                          }
                        }
                      >
                        New Goal
                      </Link>
                    </div>,
                    <div
                      className="single-button-container"
                    >
                      <input
                        className="nav-button"
                        onClick={[Function]}
                        type="submit"
                        value="Join Squad"
                      />
                    </div>,
                  ],
                  "className": "squad-dashboard-button",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className="nav-button"
                        replace={false}
                        to={
                          Object {
                            "pathname": "/CreateGoals",
                            "state": Object {
                              "userId": 1,
                            },
                          }
                        }
                      >
                        New Goal
                      </Link>,
                      "className": "single-button-container",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "New Goal",
                        "className": "nav-button",
                        "replace": false,
                        "to": Object {
                          "pathname": "/CreateGoals",
                          "state": Object {
                            "userId": 1,
                          },
                        },
                      },
                      "ref": null,
                      "rendered": "New Goal",
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <input
                        className="nav-button"
                        onClick={[Function]}
                        type="submit"
                        value="Join Squad"
                      />,
                      "className": "single-button-container",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "nav-button",
                        "onClick": [Function],
                        "type": "submit",
                        "value": "Join Squad",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
